### ISSUE TITLE
Build minimal replacement prototype updating current bucket with new devices

### ISSUE THREAD

**Issue by ashwoods** [getsentry/symx#1]
```[tasklist]
### Essential tasks
- [x] implement a downloader process that creates a stable (can be stopped and restarted at any time resumes previous downloads) mirror of `ios`, `watchos`, `tvos`, `audioos`, `accessory`, `macOS`, and `recovery` OTA images (also beta)
- [x] implement an extractor process that scans the mirror directory recursively on startup and probes each file to figure out if there is an extractable `dyld_shared_cache`, which, if found, it extracts, splits and symsorts
- [x] add stable bundle_id from the artifacts, sensible temp-dir usage, and logging in the extractor
- [x] extend the downloader also to include `IPSW` (or other) sources
- [x] extend the downloader to store a source for each image (and a hash?) so downloads can be repeated/investigated
- [x] extend the extractor with IPSW support
- [x] allow both the extractor and downloader to interact with a `GCS` bucket (either explicitly or via `gcfuse`)
- [x] setup Github Actions for cron-jobs (including GCP credentials)
```

```[tasklist]
### Nice to have
- [x] extend the extractor to (re-)run only partially (for example, specific images or specific tasks in the processing)
- [ ] ~do things in parallel? (download is net-bound, extraction is io- and compute-bound, unclear what the target is)~
```

```[tasklist]
### Out-of-scope (but unclear)
- [ ] ~validation of what we add to the symbol-store~
- [ ] ~atomic symbol store update~
- [x] ~extend the extractor to store meta-data as a result of extracting symbols (first scenario: reverse lookup from debug-id to image)~
```

Recurring topic: better differentiate the error cases in the extractor (with more input test images in the mirror). There might be different categories of errors:
- (most common?) an image doesn't contain a `dyld_shared_cache` because it is a partial update
- image cannot be decoded because we are confronted with a new encoding (either new or one we haven't seen yet)
- system error (full storage, GCS sync error, etc.)
- faulty download
- we deployed a bug
- we found a debug-id that we already stored from another image (and it has different symbols). Please take a look at the pre-symbol-store validation above.

I have only seen the first one yet, but we must react differently for each. We don't want to reprocess a file without a  `dyld_shared_cache` ever, but almost all other error cases should lead to reprocessing the failed file. There may be more variants of errors, but also in our reaction to those errors.

Another recurring topic: figure out where to store meta-data. Not all meta-data should end up in the file system, depending on how we want to index or search for it. This is particularly relevant since we want to be able to run the extractor from any machine.
